File: output/cpp/tests/transition_state_id_exec.cpp
#include <string>
#include <unordered_map>
#include <vector>
#include <any>
#include <memory>


#include <iostream>
#include <string>
#include <cassert>

class SysXFrameEvent {
public:
    std::string _message;
    std::unordered_map<std::string, std::any> _parameters;

    SysXFrameEvent(const std::string& message, std::unordered_map<std::string, std::any> params = {})
        : _message(message), _parameters(std::move(params)) {}
};

class SysXFrameContext {
public:
    SysXFrameEvent _event;
    std::any _return;
    std::unordered_map<std::string, std::any> _data;

    SysXFrameContext(SysXFrameEvent event, std::any default_return = {})
        : _event(std::move(event)), _return(std::move(default_return)) {}
};

class SysXCompartment {
public:
    std::string state;
    std::unordered_map<std::string, std::any> state_args;
    std::unordered_map<std::string, std::any> state_vars;
    std::unordered_map<std::string, std::any> enter_args;
    std::unordered_map<std::string, std::any> exit_args;
    std::unique_ptr<SysXFrameEvent> forward_event;
    std::unique_ptr<SysXCompartment> parent_compartment;

    explicit SysXCompartment(const std::string& state) : state(state) {}
};

class SysX {
private:
    std::unique_ptr<SysXCompartment> __compartment;
    std::unique_ptr<SysXCompartment> __next_compartment;
    std::vector<std::unique_ptr<SysXCompartment>> _state_stack;
    std::vector<SysXFrameContext> _context_stack;

    void __kernel(SysXFrameEvent& __e) {
        __router(__e);
        while (__next_compartment) {
            auto next_compartment = std::move(__next_compartment);
            SysXFrameEvent exit_event("<$");
            __router(exit_event);
            __compartment = std::move(next_compartment);
            if (!__compartment->forward_event) {
                SysXFrameEvent enter_event("$>");
                __router(enter_event);
            } else {
                auto forward_event = std::move(__compartment->forward_event);
                if (forward_event->_message == "$>") {
                    __router(*forward_event);
                } else {
                    SysXFrameEvent enter_event("$>");
                    __router(enter_event);
                    __router(*forward_event);
                }
            }
        }
    }

    void __router(SysXFrameEvent& __e) {
        const auto& state_name = __compartment->state;
        if (state_name == "A") {
            _state_A(__e);
        } else if (state_name == "B") {
            _state_B(__e);
        }
    }

    void __transition(std::unique_ptr<SysXCompartment> next) {
        __next_compartment = std::move(next);
    }

    void _state_A(SysXFrameEvent& __e) {
        if (__e._message == "e") {
            {
            auto __comp = std::make_unique<SysXCompartment>("B()");
            __transition(std::move(__comp));
            return;
            }
            return;
        }
    }

    void _state_B(SysXFrameEvent& __e) {
    }

public:
    SysX() {
        __compartment = std::make_unique<SysXCompartment>("A");
        SysXFrameEvent __frame_event("$>");
        __kernel(__frame_event);
    }

    void e() {
        SysXFrameEvent __e("e");
        SysXFrameContext __ctx(std::move(__e));
        _context_stack.push_back(std::move(__ctx));
        __kernel(_context_stack.back()._event);
        _context_stack.pop_back();
    }

};

// Stub functions for placeholder calls
void native() {}
void x() {}

// TAP test harness
int main() {
    printf("TAP version 14\n");
    printf("1..1\n");
    try {
        SysX s;
        s.e();
        printf("ok 1 - transition_state_id_exec\n");
    } catch (...) {
        printf("not ok 1 - transition_state_id_exec\n");
    }
    return 0;
}

File: output/go/tests/transition_state_id_exec.go

package main

import "fmt"

type SysXFrameEvent struct {
    _message    string
    _parameters map[string]any
}

type SysXFrameContext struct {
    _event  SysXFrameEvent
    _return any
    _data   map[string]any
}

type SysXCompartment struct {
    state            string
    stateArgs        map[string]any
    stateVars        map[string]any
    enterArgs        map[string]any
    exitArgs         map[string]any
    forwardEvent     *SysXFrameEvent
    parentCompartment *SysXCompartment
}

func newSysXCompartment(state string) *SysXCompartment {
    return &SysXCompartment{
        state:    state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
    }
}

func (c *SysXCompartment) copy() *SysXCompartment {
    nc := &SysXCompartment{
        state: c.state,
        stateArgs: make(map[string]any),
        stateVars: make(map[string]any),
        enterArgs: make(map[string]any),
        exitArgs:  make(map[string]any),
        forwardEvent:     c.forwardEvent,
        parentCompartment: c.parentCompartment,
    }
    for k, v := range c.stateArgs { nc.stateArgs[k] = v }
    for k, v := range c.stateVars { nc.stateVars[k] = v }
    for k, v := range c.enterArgs { nc.enterArgs[k] = v }
    for k, v := range c.exitArgs { nc.exitArgs[k] = v }
    return nc
}

type SysX struct {
    _state_stack []*SysXCompartment
    __compartment *SysXCompartment
    __next_compartment *SysXCompartment
    _context_stack []SysXFrameContext
}

func NewSysX() *SysX {
    s := &SysX{}
    s._state_stack = make([]*SysXCompartment, 0)
    s._context_stack = make([]SysXFrameContext, 0)
    s.__compartment = newSysXCompartment("A")
    s.__next_compartment = nil
    __frame_event := SysXFrameEvent{_message: "$>", _parameters: nil}
    __ctx := SysXFrameContext{_event: __frame_event, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
    return s
}

func (s *SysX) __kernel(__e *SysXFrameEvent) {
    s.__router(__e)
    for s.__next_compartment != nil {
        next_compartment := s.__next_compartment
        s.__next_compartment = nil
        exit_event := &SysXFrameEvent{_message: "<$", _parameters: s.__compartment.exitArgs}
        s.__router(exit_event)
        s.__compartment = next_compartment
        if s.__compartment.forwardEvent == nil {
            enter_event := &SysXFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
            s.__router(enter_event)
        } else {
            forward_event := s.__compartment.forwardEvent
            s.__compartment.forwardEvent = nil
            if forward_event._message == "$>" {
                s.__router(forward_event)
            } else {
                enter_event := &SysXFrameEvent{_message: "$>", _parameters: s.__compartment.enterArgs}
                s.__router(enter_event)
                s.__router(forward_event)
            }
        }
    }
}

func (s *SysX) __router(__e *SysXFrameEvent) {
    switch s.__compartment.state {
    case "A":
        s._state_A(__e)
    case "B":
        s._state_B(__e)
    }
}

func (s *SysX) __transition(next *SysXCompartment) {
    s.__next_compartment = next
}

func (s *SysX) E() {
    __e := SysXFrameEvent{_message: "E"}
    __ctx := SysXFrameContext{_event: __e, _data: make(map[string]any)}
    s._context_stack = append(s._context_stack, __ctx)
    s.__kernel(&s._context_stack[len(s._context_stack)-1]._event)
    s._context_stack = s._context_stack[:len(s._context_stack)-1]
}

func (s *SysX) _state_B(__e *SysXFrameEvent) {
}

func (s *SysX) _state_A(__e *SysXFrameEvent) {
    if __e._message == "E" {
        __compartment := newSysXCompartment("B")
        __compartment.parentCompartment = s.__compartment.copy()
        s.__transition(__compartment)
        return;
    }
}

func main() {
	fmt.Println("TAP version 14")
	fmt.Println("1..1")
	s := NewSysX()
	s.E()
	fmt.Println("ok 1 - transition_state_id_exec")
}
